test/spmd/tensor/parallel/test_parallelize_api.py

```diff
@@ -131,6 +188,62 @@ def __init__(self) -> None:
                 torch.nn.Linear(10, 5), device_mesh, PairwiseParallel()
             )
 
+    @with_comms
+    def test_linear_row_wise_parallel(self):
+        # test RowwiseParallel
+        inp_size = [8, 16]
+        rowwise = RowwiseParallel()
+
+        torch.manual_seed(5)
+        model = torch.nn.Linear(16, 10, device=self.device_type)
+        torch.manual_seed(5)
+        model_tp = torch.nn.Linear(16, 10, device=self.device_type)
+
+        # parallelize model_tp
+        device_mesh = DeviceMesh(self.device_type, list(range(self.world_size)))
+        _parallelize_linear(model_tp, device_mesh, rowwise)
+
+        LR = 0.25  # the learning rate we use for testing
+        optim = torch.optim.SGD(model.parameters(), lr=LR)
+        optim_tp = torch.optim.SGD(model_tp.parameters(), lr=LR)
+
+        torch.manual_seed(self.rank)
+        for _ in range(2):
+            inp = torch.rand(*inp_size, device=self.device_type)
+            local_inp = make_input_replicate_1d(
```

Comment:
why we need to create a DTensor then convert it to a local tensor?

Reply:
equivalent to `local_input = inp.repeat(self.world_size, 1)`. Will replace it.